swo_aws_extension/flows/steps/assign_pma.py

```diff
@@ -0,0 +1,91 @@
+from swo_aws_extension.aws.client import AWSClient
+from swo_aws_extension.aws.config import Config
+from swo_aws_extension.constants import PhasesEnum
+from swo_aws_extension.flows.order import InitialAWSContext, PurchaseContext
+from swo_aws_extension.flows.steps.base import BasePhaseStep
+from swo_aws_extension.flows.steps.errors import (
+    AlreadyProcessedStepError,
+    SkipStepError,
+    UnexpectedStopError,
+)
+from swo_aws_extension.parameters import get_phase, set_phase, set_pm_account_id
+
+logger = logging.getLogger(__name__)
+
+
+class AssignPMA(BasePhaseStep):
+    """Step to assign a Program Management Account (PMA) to the order."""
+
+    def __init__(self, config: Config, role_name: str) -> None:
+        self._config = config
+        self._role_name = role_name
+
+    def pre_step(self, context: InitialAWSContext) -> None:
+        """Hook to run before the step processing."""
+        phase = get_phase(context.order)
+        if phase != PhasesEnum.ASSIGN_PMA:
+            raise SkipStepError(
+                "%s - Next - Current phase is '%s', skipping as it is not '%s'",
+                context.order_id,
+                phase,
+                PhasesEnum.ASSIGN_PMA.value,
+            )
+
+        if context.pm_account_id:
+            context.order = set_phase(context.order, self._get_next_phase(context))
+            raise AlreadyProcessedStepError(
+                "%s - Next - PMA account %s already assigned to order %s. Continue",
+                context.order_id,
+                context.pm_account_id,
+                context.order_id,
+            )
+
+    def process(self, context: PurchaseContext) -> None:
+        """Assign PMA to the order and validate PMA credentials."""
+        pma_table = ProgramManagementAccountTable()
+        try:
+            pm_account = pma_table.get_by_authorization_and_currency_id(
+                context.authorization_id, context.currency
+            )
+        except AirtableRecordNotFoundError:
+            raise UnexpectedStopError(
+                f"No Program Management Account found for Authorization ID "
+                f"{context.authorization_id}",
+                "Failed to find a Program Management Account for "
+                f"Authorization ID {context.authorization_id} and currency {context.currency}. "
+                "Please ensure that a valid PMA exists in Airtable.",
+            )
+
+        context.aws_client = AWSClient(self._config, pm_account.pma_account_id, self._role_name)
+        if not context.aws_client.are_credentials_valid():
+            raise UnexpectedStopError(
+                f"Program Management Account {pm_account.pma_account_id} failed to retrieve "
+                f"credentials",
+                f"The Program Management Account {pm_account.pma_account_id} is failing "
+                f"to provide valid AWS credentials. Please verify that the account ID is correct",
+            )
```

Comment:
(not needed now) Umm...I'm thinking we should extract this logic from this step to avoid repeating it every time we need an AWSClient. should it part of the context, a dedicated step, or maybe a separate service layer that knows how to instantiate the aws_client and exposes the methods the step needs?

The idea would be:

* Step: don't know anything about how AWS Client works)
* AWSService: knows how instantiate the client, contains the AWS logic, and expose methods
* AWSClient